unity-renderer/Assets/UIComponents/Tests/SectionSelectorComponentViewTests.cs

```diff
@@ -0,0 +1,145 @@
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.TestTools;
+using UnityEngine.UI;
+
+public class SectionSelectorComponentViewTests
+{
+    private SectionSelectorComponentView sectionSelectorComponent;
+
+    [SetUp]
+    public void SetUp() { sectionSelectorComponent = BaseComponentView.Create<SectionSelectorComponentView>("SectionSelector"); }
+
+    [TearDown]
+    public void TearDown()
+    {
+        sectionSelectorComponent.Dispose();
+        GameObject.Destroy(sectionSelectorComponent.gameObject);
+    }
+
+    [Test]
+    public void ConfigureSectionSelectorCorrectly()
+    {
+        // Arrange
+        SectionSelectorComponentModel testModel = CreateTestModel(3);
+
+        // Act
+        sectionSelectorComponent.Configure(testModel);
+
+        // Assert
+        Assert.AreEqual(testModel, sectionSelectorComponent.model, "The model does not match after configuring the button.");
+    }
+
+    [Test]
+    public void RefreshSectionSelectorCorrectly()
+    {
+        // Arrange
+        sectionSelectorComponent.model = CreateTestModel(3);
+        sectionSelectorComponent.model.sections.Add(new SectionToggleModel
+        {
+            icon = Sprite.Create(new Texture2D(20, 20), new Rect(), Vector2.zero),
```

Comment:
The `Texture2D` objects have to be destroyed manually to avoid leaks.

Reply:
Good catch! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/531831692da512ed2d0095f4e2c0d08be70ee7be